cmd/tsidp/tsidp.go

```diff
@@ -345,7 +346,8 @@ func (ar *authRequest) allowRelyingParty(r *http.Request, lc *local.Client) erro
 			clientID = r.FormValue("client_id")
 			clientSecret = r.FormValue("client_secret")
 		}
-		if ar.funnelRP.ID != clientID || ar.funnelRP.Secret != clientSecret {
+		if subtle.ConstantTimeCompare([]byte(clientID), []byte(ar.funnelRP.ID)) != 1 ||
```

Comment:
I suppose if we really want it to be constant time we should do both comparisons before checking their results, but this matches what was there.